lib/logger.js

```diff
@@ -1,9 +1,17 @@
-const { evolve, is, pick, pipe, tap, when } = require('ramda')
+const {
+  converge, evolve, identity, is, merge, pick, pipe, tap, when
+} = require('ramda')
 
-module.exports =
+const clean =
+  pick(['message', 'name', 'stack'])
+
+const flattenProps =
+  converge(merge, [ identity, clean ])
```

Comment:
😕 I don't follow. Are we picking the props off of an object & then merging them back into the same object?

Reply:
Javascript errors are super weird.  The `{ message, name, stack }` properties are not "own" properties.  Instead they are built-in getters.  Consequently:
```js
JSON.stringify(new Error('my message')) === '{}'
```
Which is not at all what we want.  In the past we've resolved this with a `pick`, which reads the specified props whether they are "own" or not.

With the code in this PR, if we've attached new "own" properties to an error, it will `pick` off the special `{ message, name, stack }` props, and then merge them with the extra "own" props, flattening them into a serializable object.

Reply:
Oooh... okay. A comment may be in order.

Reply:
Maybe "own" is the wrong word.  I may be thinking of "enumerable".  Those three props are not "enumerable", which is why a `JSON.stringify` won't pick them up when serializing.